integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/foreach/ForEachPropertiesTestCase.java

```diff
@@ -101,28 +101,21 @@ public void testSingleForEachProperties() throws Exception {
             assertTrue(carbonLogReader.getLogs().contains("in_count = " + 3),
                     "Final counter mismatch, expected 3 found = " + carbonLogReader.getLogs());
         }
-
-        if (carbonLogReader.checkForLog("in_payload", DEFAULT_TIMEOUT)) {
-            //final payload in insequence
-            String payload = carbonLogReader.getLogs();
-            String search = "<m0:getQuote>(.*)</m0:getQuote>";
-            Pattern pattern = Pattern.compile(search, Pattern.DOTALL);
-            Matcher matcher = pattern.matcher(payload);
-            boolean matchFound = matcher.find();
-
-            assertTrue(matchFound, "getQuote element not found");
-            if (matchFound) {
-                int start = matcher.start();
-                int end = matcher.end();
-                String quote = payload.substring(start, end);
-                assertTrue(quote.contains("<m0:group>Group1</m0:group>"), "Group Element not found in : " + quote);
-                assertTrue(quote.contains("<m0:symbol>Group1_IBM</m0:symbol>"), "IBM Element not found in : " + quote);
-                assertTrue(quote.contains("<m0:symbol>Group1_WSO2</m0:symbol>"),
-                           "WSO2 Element not found in : " + quote);
-                assertTrue(quote.contains("<m0:symbol>Group1_MSFT</m0:symbol>"),
-                           "MSTF Element not found in : " + quote);
-            }
-        }
+        //final payload in insequence
+        String payload = carbonLogReader.getLogs();
+        String search = "<m0:getQuote>(.*)</m0:getQuote>";
+        Pattern pattern = Pattern.compile(search, Pattern.DOTALL);
+        Matcher matcher = pattern.matcher(payload);
+        boolean matchFound = matcher.find();
+
+        assertTrue(matchFound, "getQuote element not found");
+        int start = matcher.start();
+        int end = matcher.end();
+        String quote = payload.substring(start, end);
+        assertTrue(quote.contains("<m0:group>Group1</m0:group>"), "Group Element not found in : " + quote);
+        assertTrue(quote.contains("<m0:symbol>Group1_IBM</m0:symbol>"), "IBM Element not found in : " + quote);
+        assertTrue(quote.contains("<m0:symbol>Group1_WSO2</m0:symbol>"), "WSO2 Element not found in : " + quote);
+        assertTrue(quote.contains("<m0:symbol>Group1_MSFT</m0:symbol>"), "MSTF Element not found in : " + quote);
```

Comment:
I see that we have only removed the outer if condition. Even though we've removed the inner if (matchFound) {, we still have the assert, which I feel like is still the same as before. Am I missing something?

Reply:
Yes the inner asserts will fail if it reach there since we are not logging the payload. Hence added a log level full in the artifact to fix this case as mentioned in the PR description.